### PR TITLE
[develop2] Cache clean

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: Feature: Add `conan cache clean` subcommand to clean build and source folders.
Close https://github.com/conan-io/conan/issues/11562

### CODE REVIEW

**Review by SSE4 — COMMENTED** [review]
is it safe? what happens if it raises in the middle of removal, will it leave some incomplete or inconsistent cache state? (or user just hits Ctrl-C/Ctrl-Break in the middle of the command).

File: conan/cli/commands/cache.py [hunk 1]
```diff
@@ -48,3 +49,26 @@ def cache_path(conan_api: ConanAPI, parser, subparser, *args):
         else:
             raise ConanException(f"'--folder {args.folder}' requires a recipe reference")
     return path
+
+
+@conan_subcommand(formatters={"text": default_text_formatter})
```
Comment (czoido) [inline conan/cli/commands/cache.py]: If the command has no output to console we don't need to have the text formatter.
```suggestion
@conan_subcommand()
```

**Review by czoido — APPROVED** [review]
Just the comment about the formatter and that the command required at least one argument saying what you want to clean.

**Comment by czoido** [comment]
> is it safe? what happens if it raises in the middle of removal, will it leave some incomplete or inconsistent cache state? (or user just hits Ctrl-C/Ctrl-Break in the middle of the command).

@SSE4 I think this should be safe as we are not touching the database, if something goes wrong, a second execution would fix the  clean that did not finish.

File: conan/cli/commands/cache.py [hunk 2]
```diff
@@ -48,3 +49,29 @@ def cache_path(conan_api: ConanAPI, parser, subparser, *args):
...
+                           help="Remove all packages (empty) or provide a query: "
+                                "os=Windows AND (arch=x86 OR compiler=gcc)")
+    args = parser.parse_args(*args)
+
+    if not args.source and not args.build and not args.download:
+        raise ConanException("Define at least one argument from --source, --build, --download")
+
+    ref_pattern = ListPattern(args.pattern, rrev="*", package_id="*", prev="*")
+    package_list = conan_api.list.select(ref_pattern, package_query=args.package_query)
+    conan_api.cache.clean(package_list, source=args.source, build=args.build,
```
Comment (AbrilRBS) [inline conan/cli/commands/cache.py:76]: Love how clean this looks

**Review by AbrilRBS — APPROVED** [review]
👍

**Pull request merged**